ksqldb-parser/src/main/java/io/confluent/ksql/parser/tree/WithinExpression.java

```diff
@@ -107,13 +134,18 @@ public boolean equals(final Object o) {
     final WithinExpression withinExpression = (WithinExpression) o;
     return before == withinExpression.before && after == withinExpression.after
            && Objects.equals(beforeTimeUnit, withinExpression.beforeTimeUnit)
-           && Objects.equals(afterTimeUnit, withinExpression.afterTimeUnit);
+           && Objects.equals(afterTimeUnit, withinExpression.afterTimeUnit)
+           && Objects.equals(gracePeriod, withinExpression.gracePeriod);
   }
 
   private JoinWindows createJoinWindows() {
     final JoinWindows joinWindow = JoinWindows
-        .of(Duration.ofMillis(beforeTimeUnit.toMillis(before)));
-    return joinWindow.after(Duration.ofMillis(afterTimeUnit.toMillis(after)));
+        .of(Duration.ofMillis(beforeTimeUnit.toMillis(before)))
+        .after(Duration.ofMillis(afterTimeUnit.toMillis(after)));
+
+    return (gracePeriod.isPresent())
+        ? joinWindow.grace(gracePeriod.get().toDuration())
```

Comment:
This is actually "incorrect". We should apply GRACE PERIOD only via the new `JoinWindows#ofSizeAndGrace()` and never call `grace()` directly. -- Of course, we first need to get the AK PR merged. 

Just wanted to point it out, that we need to change this in a follow up PR. (Wondering though, we we need to apply GRACE in this PR at all? Could we not also just ignore it when building the plan until we can do it properly? Or do we need it for some testing?)

Reply:
There's another jira is planned to call the new API, so a follow-up PR will include it once the AK PR is merged. Another thing is that this part of the code, which creates the `JoinWindows`, does not actually do anything with grace in the Kafka streams yet. Later on, another part of the code just gets the after/before windows from this JoinWindows, but not the grace period yet. This is just a preparation for the WithinExpression.